server/shared/src/main/scala/org/http4s/server/websocket/WebSocketBuilder2.scala

```diff
@@ -47,6 +48,8 @@ sealed abstract class WebSocketBuilder2[F[_]: Applicative] private (
 ) {
   import WebSocketBuilder2.impl
 
+  private[http4s] val wsKey: Key[WebSocketContext[F]] = webSocketKey
```

Comment:
Don't know what you guys think about this, could have made it `private[http4s] val webSocketKey` directly in the `sealed abstract class` arguments, but that could probably affect binary compatibility? 

The idea is to have this accessible from `EmberServerBuilder` as there is a key per WS builder.

Reply:
@gvolpe would you mind adjusting this one as well?

Reply:
Sure, no problems. Make it a `private[http4s] val webSocketKey` then?

Reply:
Actually, I wonder if it should be completely public. Otherwise you can't implement a websocket server outside of `org.http4s`?

Reply:
You cannot do it at the moment AFAIU, but that's deeply rooted in the code: [WebSocket](https://github.com/http4s/http4s/blob/series/0.23/core/shared/src/main/scala/org/http4s/websocket/WebSocket.scala#L22)

I don't think making `sealed abstract class WebSocketBuilder2[F[_]: Applicative] private (...)` public makes any difference at the moment, but I'm not sure how I feel about making this argument a public val... No preferences, I guess.

Reply:
I landed at that wall when I was trying implement one myself 🥲

Reply:
Ah, didn't realize that. `private[http4s]` makes sense then.